liquibase-core/src/main/java/liquibase/lockservice/ProlongingLockService.java

```diff
@@ -0,0 +1,879 @@
+            database.commit();
+        }
+
+        if (executor.updatesDatabase() && (database instanceof DerbyDatabase) && ((DerbyDatabase) database)
+            .supportsBooleanDataType()) {
+            //check if the changelog table is of an old smallint vs. boolean format
+            Object obj = executor.queryForObject(
+                new RawSqlStatement(
+                    "SELECT MIN(locked) AS test FROM " + databaseLockTableName() + " FETCH FIRST " +
+                        "ROW ONLY"
+                ), Object.class
+            );
+            if (!(obj instanceof Boolean)) { //wrong type, need to recreate table
+                executor.execute(
+                    new DropTableStatement(
+                        database.getLiquibaseCatalogName(),
+                        database.getLiquibaseSchemaName(),
+                        database.getDatabaseChangeLogLockTableName(),
+                        false
+                    )
+                );
+                executor.execute(new CreateDatabaseChangeLogLockTableStatement());
+                executor.execute(new InitializeDatabaseChangeLogLockTableStatement());
+            }
+
+            database.commit();
+        }
+
+        checkAndUpdateLockTableWithLockExpiresAndById(executor);
+
+    }
+
+    /**
+     * If the table does not already have the LOCKEXPIRES and LOCKEDBYID columns, upgrade.
+     *
+     * @param executor
+     * @throws DatabaseException
+     */
+    private void checkAndUpdateLockTableWithLockExpiresAndById(Executor executor)
+        throws DatabaseException {
+
+        if (newColumnsExist(executor)) {
+            return;
+        }
+
+        AddColumnStatement addLockExpires =
+            new AddColumnStatement(database.getLiquibaseCatalogName(),
+                database.getLiquibaseSchemaName(),
+                database.getDatabaseChangeLogLockTableName(),
+                "LOCKEXPIRES",
+                getDateTimeTypeString(database),
+                null);
+
+        AddColumnStatement addLockedById =
+            new AddColumnStatement(database.getLiquibaseCatalogName(),
+                database.getLiquibaseSchemaName(),
+                database.getDatabaseChangeLogLockTableName(),
+                "LOCKEDBYID",
+                getCharTypeName(database) + "(36)",
+                null);
+
+        executor.execute(addLockExpires);
+        executor.execute(addLockedById);
+
+        database.commit();
+    }
+
+    protected String getDateTimeTypeString(Database database) {
+        if (database instanceof MSSQLDatabase) {
+            try {
+                if (database.getDatabaseMajorVersion() >= 10) { // 2008 or later
+                    return "datetime2(3)";
+                }
+            } catch (DatabaseException e) {
+                // ignore
+            }
+        }
+        return "datetime";
+    }
+
+    private boolean newColumnsExist(Executor executor) throws DatabaseException {
```

Comment:
"newColumns" isn't a great method name here: the columns are new now, but the next time this class is modified they won't be, and this method name will be obsolete and confusing. Better to name more like `lockProlongColumnsExist` or similar IMO.

Reply:
True, agree. Again, was just a POC, we would have to think how liquibase manages their own database schema...
maybe with liquibase? :D